### DISCUSSION TITLE
Using Javascript Promise instead of Yield

### DISCUSSION THREAD

**Discussion by abdollahkahne** [dotnet/AspNetCore.Docs#24286] [Q&A]
https://github.com/dotnet/AspNetCore.Docs/blob/b0a9a7d152c29af429ebf01e24a6a3c55be800d9/aspnetcore/signalr/streaming/samples/3.0/wwwroot/js/stream.js#L42
As a reader I could not understand and connect with `yield` here. I presume, `yield` is used here to be sure that it `fulfilled` before continuing. May be its better to use the **promise** `then` and `catch` block here from my point of view as I do it here for example. And maybe some people be easier with `yield` version. I just want to share it here. Sorry if it may be boring:

```
function handleSubject(subject) {
            var iteration=0;
            const intervalHandle=setInterval(() => {
                iteration++;
                subject.next(iteration.toString());
                if (iteration === 20) {
                    clearInterval(intervalHandle);
                    subject.complete();
                }
            }, 1500);
}
```
```
// build hub connection
const hubConnection=new signalR.HubConnectionBuilder()
        .withUrl("/streaming").configureLogging(signalR.LogLevel.Information).build();

// start hub as promise and then invoke methods after starting connection
hubConnection.start().then(()=> {
            const mySubject=new signalR.Subject();
            hubConnection.send("UploadStream",mySubject)
            .then(()=>{handleSubject(mySubject);})
            .catch(err =>{console.error(err)});
});
```

**Comment by abdollahkahne** [comment]
I was not familiar with generators. May be because of similarities here between generator and enumerators `yield` be better choice

**Reply by BrennanConroy** [reply]
The javascript file is generated from the typescript file at https://github.com/dotnet/AspNetCore.Docs/blob/b0a9a7d152c29af429ebf01e24a6a3c55be800d9/aspnetcore/signalr/streaming/samples/3.0/wwwroot/ts/stream.ts

It might be nicer to just write the javascript file with `.then()` instead of compiling ts -> js.
